easybuild/framework/easyblock.py

```diff
@@ -1585,8 +1561,9 @@ def post_iter_step(self):
 
     def det_iter_cnt(self):
         """Determine iteration count based on configure/build/install options that may be lists."""
-        iter_opt_counts = [len(self.cfg[opt]) for opt in ITERATE_OPTIONS
-                           if opt not in ['builddependencies'] and isinstance(self.cfg[opt], (list, tuple))]
+        # Using get_ref to avoid template substitution (and possible failures)
```

Comment:
> since the failures are failures in resolving template values, which are actually just warnings

I'd actually propose to promote those to hard errors. As this issue/PR shows a failure in resolving a template is usually something not wanted and/or points to an issue in the code and/or EC. Warnings can be (and usually are) ignored as they are buried in a huge log.

Reply:
There are situations where values are picked up when the corresponding template value is not defined yet (see for example https://github.com/easybuilders/easybuild-framework/pull/3160), so if we promote this to hard errors, we'd have to do it very carefully to avoid breaking stuff that has worked fine until now.

Reply:
For some definition of "worked fine". Each template should be resolved once and with all values available or the result is *very likely* unexpected or wrong. So those breaking ECs are likely rare or faulty.

For some datapoint: I built a large chunk of ECs with this promoted to hard error to detect issues fixed here.

Proposal for road forward: Promote to hard errors on CI and show a big fat deprecation warning on regular usages if templating fails. As mentioned there may only be a few affected ECs/EBs if any and this way no new failing ones can be added.

Reply:
That makes a lot of sense.

Can you look into a separate PR that adds a configuration option to promote template failures to a hard error, so we can enable it in the easyconfigs test suite and in our regression testing setup?

Please make the suggested change for the comment here though, so we can get this PR merged...

Reply:
> Please make the suggested change for the comment here though, so we can get this PR merged...

If it is a goal to make it an error (or better: enforce correct substition and treat failure as an error), why change that comment? Isn't it correct as-is? May I suggest `using get_ref to avoid resolving templates as their required attributes may not be available yet`?

Reply:
I'm OK with what you suggest, I'm mostly concerned about the `(and possible failures)` (since nothing fails, you just get warnings, which usually actually are OK in this case).